docs/.vitepress/vitepress/components/doc-content/vp-table-of-content.vue

```diff
@@ -24,20 +52,31 @@ const sponsor = computed(() => sponsorLocale[lang.value])
       <h3 class="toc-content__heading">Contents</h3>
       <ul class="toc-items">
         <li
-          v-for="{ link, text, children } in headers"
+          v-for="{ link, text, children, tag, className } in convertHeaders"
           :key="link"
           class="toc-item"
         >
-          <a class="toc-link" :href="link" :title="text">{{ text }}</a>
+          <a class="toc-link" :href="link" :title="text">
+            {{ text }}
+            <span v-if="tag" :class="className">{{ tag }}</span>
+          </a>
           <ul v-if="children">
             <li
-              v-for="{ link: childLink, text: childText } in children"
+              v-for="{
+                link: childLink,
+                text: childText,
+                tag: childTag,
+                className: childClassName,
+              } in children"
               :key="childLink"
               class="toc-item"
             >
-              <a class="toc-link subitem" :href="childLink" :title="text">{{
-                childText
```

Comment:
Using the markdown plugin to parse content may be a better idea.

[refer to localMd](https://github.com/element-plus/element-plus/blob/dev/docs/.vitepress/config/plugins.ts#L15)